frontend/src/components/Sidebar/Sidebar.tsx

```diff
@@ -64,7 +68,9 @@ const SidebarComponent: React.FC<Props> = ({ schedule, major }) => {
 
 const mapStateToProps = (state: AppState) => ({
   schedule: getScheduleFromState(state),
-  major: getDeclaredMajorFromState(state),
+  major: getMajors(state).find(
+    (major: Major) => major.name === getScheduleDataFromState(state).major
+  ),
```

Comment:
The schedule state only retains major as a string, while the sidebar requires a `Major`. This just retrieves the major from the stored `Major`s based upon what the string stored in the schedule is.